bionic/gcs.py

```diff
@@ -37,99 +35,33 @@ def get_gcs_client_without_warnings(cache_value=True):
         warnings.filterwarnings(
             "ignore", "Your application has authenticated using end user credentials"
         )
-        logger.info("Initializing GCS client ...")
-        return gcs.Client()
-
-
-def copy_to_gcs(src, dst):
-    """Copy a local file at src to GCS at dst"""
-    bucket = dst.replace("gs://", "").split("/")[0]
-    prefix = f"gs://{bucket}"
-    path = dst[len(prefix) + 1 :]
+        logger.info("Initializing GCS filesystem ...")
+        return fsspec.filesystem("gcs")
 
-    client = get_gcs_client_without_warnings()
-    blob = client.get_bucket(bucket).blob(path)
-    blob.upload_from_filename(src)
 
+# TODO: Consider using persistence.GcsFilesystem instead of exposing this function.
+def upload_to_gcs(path, url):
+    """Copy a local path to GCS url
 
-class GcsTool:
+    This method is a proxy for _upload_to_gcs which does the actual work.
+    The proxy exists so that _upload_to_gcs can be replaced for testing.
     """
-    A helper object providing utility methods for accessing GCS.  Maintains
-    a GCS client, and a prefix defining a default namespace to read/write on.
-    """
-
-    _GS_URL_PREFIX = "gs://"
-
-    def __init__(self, url):
+    _upload_to_gcs(path, url)
+
+
+def _upload_to_gcs(path, url):
+    fs = get_gcs_fs_without_warnings()
+    if path.is_dir():
+        fs.put(str(path), url, recursive=True)
+    else:
+        # If gcs url is a folder, we want to write the file in the folder.
```

Comment:
nit: "GCS URL" and "`put` API" should have the word "the" in front of them. In general, every noun needs an article unless it's a proper name, and these are not.